File: jphp-javafx-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/paint/UXColor.java
package org.develnext.jphp.ext.javafx.classes.paint;

import javafx.scene.paint.Color;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.lang.BaseWrapper;
import php.runtime.reflection.ClassEntity;

@Reflection.Name(JavaFXExtension.NAMESPACE + "paint\\UXColor")
public class UXColor extends BaseWrapper<Color> {
    interface WrappedInterface {
        @Property double red();
        @Property double blue();
        @Property double green();
        @Property double opacity();
        @Property double brightness();
        @Property double hue();
        @Property double saturation();

        Color grayscale();
        Color invert();
        Color saturate();
        Color desaturate();
        Color interpolate(Color endValue, double t);
    }

    public UXColor(Environment env, Color wrappedObject) {
        super(env, wrappedObject);
    }

    public UXColor(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct(double red, double blue, double green) {
        __wrappedObject = new Color(red, blue, green, 1.0);
    }

    @Signature
    public void __construct(double red, double blue, double green, double opacity) {
        __wrappedObject = new Color(red, blue, green, opacity);
    }

    @Signature
    public static Color of(String colorString) {
        return Color.web(colorString);
    }

    @Signature
    public static Color of(String colorString, double opacity) {
        return Color.web(colorString, opacity);
    }

    @Signature
    public static Color rgb(int red, int blue, int green) {
        return Color.rgb(red, blue, green);
    }

    @Signature
    public static Color rgb(int red, int blue, int green, double opacity) {
        return Color.rgb(red, blue, green, opacity);
    }
}

File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/paint/UXColor.php
<?php
namespace php\gui\paint;

/**
 * Class UXColor
 * @package php\gui\paint
 * @packages gui, javafx
 */
class UXColor
{
    /**
     * Уровень красного (от 0 до 1)
     * @readonly
     * @var double
     */
    public $red;

    /**
     * Уровень синего (от 0 до 1).
     * @readonly
     * @var double
     */
    public $blue;

    /**
     * Уровень зеленого (от 0 до 1).
     * @readonly
     * @var double
     */
    public $green;

    /**
     * Уровень прозрачности (от 0 до 1).
     * @readonly
     * @var double
     */
    public $opacity;

    /**
     * Уровень яркости.
     * @readonly
     * @var double
     */
    public $brightness;

    /**
     * @readonly
     * @var double
     */
    public $hue;

    /**
     * @readonly
     * @var double
     */
    public $saturation;

    /**
     * WEB значение цвета.
     * @readonly
     * @var string
     */
    public $webValue;

    /**
     * @param double $r
     * @param double $g
     * @param double $b
     * @param double $opacity (optional)
     */
    public function __construct($r, $g, $b, $opacity) {}

    /**
     * @return UXColor
     */
    public function grayscale() {}

    /**
     * @return UXColor
     */
    public function invert() {}

    /**
     * @return UXColor
     */
    public function saturate() {}

    /**
     * @return UXColor
     */
    public function desaturate() {}

    /**
     * @param UXColor $color
     * @param double $t
     * @return UXColor
     */
    public function interpolate(UXColor $color, $t) {}

    /**
     * @return int
     */
    public function getRGB()
    {
    }

    /**
     * @return string
     */
    public function getWebValue()
    {
    }

    /**
     * @param string $colorString
     * @return UXColor
     */
    public static function of($colorString) {}

    /**
     * @param int $r
     * @param int $g
     * @param int $b
     * @param double $opacity [optional]
     * @return UXColor
     */
    public static function rgb($r, $g, $b, $opacity) {}
}